File: mahaam-api-java/src/main/java/mahaam/infra/monitor/AuditController.java
package mahaam.infra.monitor;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import jakarta.ws.rs.FormParam;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.core.Response;
import mahaam.infra.Config;
import mahaam.infra.Log;

public interface AuditController {
	Response error(String error);

	Response trace(String info);
}

@ApplicationScoped
@Path("/audit")
class DefaultAuditController implements AuditController {

	@Inject
	Config config;

	@POST
	@Path("/error")
	@Override
	public Response error(@FormParam("error") String error) {
		Log.error("mahaam-mb: " + error);
		return Response.status(Response.Status.CREATED).build();
	}

	@POST
	@Path("/info")
	@Override
	public Response trace(@FormParam("info") String info) {
		Log.info("mahaam-mb: " + info);
		return Response.status(Response.Status.CREATED).build();
	}
}

File: mahaam-api-ts/src/infra/monitor/audit.controller.ts
import { Controller, Post, Body, Res } from '@nestjs/common';
import { Response } from 'express';
import Log from '../log';

export interface AuditController {
  error(error: string, res: Response): Promise<void>;
  info(info: string, res: Response): Promise<void>;
}

@Controller('audit')
export class DefaultAuditController implements AuditController {
  @Post('error')
  async error(@Body('error') error: string, @Res() res: Response): Promise<void> {
    Log.error('mahaam-mb: ' + error);
    res.sendStatus(201);
  }

  @Post('info')
  async info(@Body('info') info: string, @Res() res: Response): Promise<void> {
    Log.info('mahaam-mb: ' + info);
    res.sendStatus(201);
  }
}
